### PR TITLE
[SPARK-20535][SPARKR] R wrappers for explode_outer and posexplode_outer

### PR DESCRIPTION
**Pull request by zero323**
## What changes were proposed in this pull request?

Ad R wrappers for

- `o.a.s.sql.functions.explode_outer`
- `o.a.s.sql.functions.posexplode_outer`

## How was this patch tested?

Additional unit tests, manual testing.

### CODE REVIEW

File: R/pkg/R/functions.R [hunk 1]
```diff
@@ -3803,3 +3803,58 @@ setMethod("repeat_string",
...
+          signature(x = "Column"),
+          function(x) {
+            jc <- callJStatic("org.apache.spark.sql.functions", "explode_outer", x@jc)
+            column(jc)
+          })
+
+#' posexplode_outer
+#'
+#' Creates a new row for each element with position in the given array or map column.
+#' Unlike \code{posexplode}, if the array/map is null or empty then the row (null, null) is produced.
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: do you want to put null in \code as L3810
typically I'd say `NULL` instead but this is actually null on JVM side so I think it's ok as it is/

**Review by felixcheung — APPROVED** [review]
looks good

**Comment by zero323** [comment]
Jenkins retest this please.

**Comment by zero323** [comment]
Jenkins retest this please.

**Comment by felixcheung** [comment]
merged to master, thanks!

**Comment by zero323** [comment]
Thanks @felixcheung

**Pull request closed without merging**